libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -174,6 +174,17 @@ void ReferencesResolver::endVisit(ModifierDefinition const&)
 void ReferencesResolver::endVisit(IdentifierPath const& _path)
 {
 	Declaration const* declaration = m_resolver.pathFromCurrentScope(_path.path());
```

Comment:
> I think we should first fix reference resolution inside a struct to not consider the members at all

Looking at the comment in the issue.

So shouldn't this be fixed inside `pathFromCurrentScope`?

Reply:
I am not sure, I would say that semantics is the same.

Reply:
Yeah. The only other use case seems to be for NatSpec, where `Enum` would not matter. So the semantics should be the same. But I think it's better to have the code for special case inside `pathFromCurrentScope`, in case `pathFromCurrentScope` needs to be used at a different place. What do you think?

Reply:
I think it is a good idea, which also helps to reduce the amount of changes :).